common/util.go

```diff
@@ -128,9 +130,34 @@ func BytesToVideoProfile(txData []byte) ([]ffmpeg.VideoProfile, error) {
 	return profiles, nil
 }
 
+func FFmpegProfiletoNetProfile(ffmpegProfiles []ffmpeg.VideoProfile) ([]*net.VideoProfile, error) {
+	profiles := []*net.VideoProfile{}
+	for _, profile := range ffmpegProfiles {
+		width, height, err := ffmpeg.VideoProfileResolution(profile)
+		if err != nil {
+			return nil, err
+		}
+		br := strings.Replace(profile.Bitrate, "k", "", 1)
+		bitrate, err := strconv.Atoi(br)
+		if err != nil {
+			return nil, err
+		}
+		if profile.Name == "" {
+			profile.Name = fmt.Sprintf("%dx%d_%d", width, height, bitrate)
```

Comment:
Should the `fmt.Sprintf` part be factored out into a helper for reuse? That would allow us to catch all instances where we need to change the format for whatever reason. Also I'm a little nervous about mutating the profile itself when it's not strictly necessary. Can an intermediate `name` variable be used instead?

One other trick that might come in handy is to also prefix (or append) the generated name with *where* it came from, so we know at what point the name was missing. Something like this:

```
func defaultProfileName(w, h, b) {
  return fmt.Sprintf(...)
}

// in createRTMPStreamIDHandler
name = "webhook_"+defaultProfileName(...)

// in FFmpegProfiletoNetProfile
name = "ffmpeg_"+defaultProfileName(...)

// in makeFfmpegVideoProfiles
name = "net_"+defaultProfileName(...)
```

Reply:
How can "mutating the profile" cause issues down the line? Cool, I'll add the prefix!

Reply:
> How can "mutating the profile" cause issues down the line

In this specific instance it is probably mostly OK since the struct being mutated isn't used anywhere else, but in general it's much safer to avoid side effects where possible. Issues include concurrency, reasoning about effects outside the scope of the function, etc.

So I'd keep the entire struct immutable here as good practice, if only to future-proof things. Of course, this only works if future maintainers also uphold the same discipline. For a language of its age, I'm really surprised Golang went with mutable-by-default variables.

Reply:
Thanks for the explanation! I do remember having issues with that at one point